src/lib/adapters/SendgridAdapter.ts

```diff
@@ -13,7 +14,7 @@ export const createSendgrid = (adapterOptions: SendgridOptions): Adapter<Sendgri
         options: adapterOptions,
         mailService: sendgrid,
         sender: (sendOptions) => {
-            const send: AdapterSend<Partial<MailData>> = (message, options) => {
+            const send: AdapterSend<Partial<SendgridMessage>> = (message, options) => {
```

Comment:
NIT: This change probably slipped by and is likely not related to the PR. No need to rebase tho.

Reply:
yes i know, I needed to add this message type to the nodemailer so i did the same thing here without creating the new PR